Cleanup Python Examples

Mani-Sadhasivam opened the pull request:
This PR cleans-up the Python examples and fixes the Python docs.

Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>

Mani-Sadhasivam commented:
@arfoll Any update on this PR?

alext-mkrs commented on examples/python/gpio_advanced.py:
```diff
@@ -21,34 +21,39 @@
-  c.count+=1
+def isr_routine(gpio):
+    print("pin " + repr(gpio.getPin(True)) + " = " + repr(gpio.read()))
+    c.count+=1
```
While we are at it, this needs spaces around `+=`

Mani-Sadhasivam replied:
Okay. Will add spaces around `=` in all examples

alext-mkrs commented on examples/python/gpio_advanced.py:
```diff
@@ -21,34 +21,39 @@
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE
+#
+# Example Usage: Reads GPIO 6 through interrupt
```
This description is IMHO a bit misleading. The idea of this functionality and the example is not in the specific fact we *read* the GPIO when the interrupt happens, but in the general fact is that we can *execute a predefined piece of code upon GPIO state change*, so I'd suggest highlighting namely that.

Mani-Sadhasivam replied:
Okay. How about `Trigger ISR upon GPIO state change`?

alext-mkrs replied:
Yep, sounds good.

alext-mkrs commented on examples/python/i2c_bmp85.py:
```diff
@@ -21,20 +21,21 @@
-  print("error")
+    print("error")
 
-# we want to read temperature so write 0x2e into control reg
```
I think this comment actually makes sense, so tha tpeople understand what the example does with the sensor, without looking into the datasheet. Why do you think it needs to be removed?

Mani-Sadhasivam replied:
Hmm. I thought these examples are here to _just_ demonstrate the basic usage and not provide much of the device specific details. However, I agree with you. Having this line will help the user without looking into datasheet. I will keep it as it is!

alext-mkrs commented on examples/python/led.py:
```diff
@@ -0,0 +1,53 @@
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# Example Usage: Reads maximum brightness value for user1 led and turns it
+# 	             on/off depending on current state. Then sets led trigger
```
The indentation is off on this line

Mani-Sadhasivam replied:
Ack

alext-mkrs commented on examples/python/led.py:
```diff
@@ -0,0 +1,53 @@
+
+print("maximum brightness value for user1 led is: %d" % val);
+
+if (val >= 1):
```
It would probably be helpful to clarify the logic of this piece in a comment.

Mani-Sadhasivam replied:
Okay. Will add a comment.

alext-mkrs commented on examples/python/spi_loopback.py:
```diff
@@ -21,20 +21,23 @@
+    if rxbuf != txbuf:
+        print("Data mismatch!")
+        break
+        exit(1)
```
Unreachable code. I know it was in the original example, but let's just fix that while we are at it. I'd suggest removing the `break` and leaving just the `exit(1)`, that would be in line with my fix for other example.

Mani-Sadhasivam replied:
Ack

alext-mkrs commented on examples/python/uart_sender.py:
```diff
@@ -21,6 +21,10 @@
 # OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE
+#
+# Example Usage: Sends data through UART and expects `X` flag as response from
```
This description is incorrect, `_sender` doesn't expect to receive any flags, it *sends* a bunch of stuff, including a flag and then just reads and displays whatever data it receives after that.

Mani-Sadhasivam replied:
Err. This was my bad. How about, `Sends some info messages along with X flag and expects a response from the other end. This example requires uart_receiver.py to be running on the other end`?

Mani-Sadhasivam commented:
@alext-mkrs Incorporated your review suggestions. Below is the changeset:

1. Squashed both commits into one
2. Added spaces around `=` and `+=`
3. Modified and added some comments based on review

Mani-Sadhasivam commented:
I'm not sure of the conflict appearing now :(

alext-mkrs commented:
Those conflicts are because of my recent fixes in those files, you just need to rebase to the current master.

alext-mkrs commented:
@Mani-Sadhasivam, BTW, that conflict is not a problem, I can fix that by myself upon merging, but let me see if you'd want to train your git-fu by fixing that yourself ;)

Mani-Sadhasivam commented:
@alext-mkrs Rebased on top of master!

alext-mkrs commented on examples/python/spi.py:
```diff
@@ -22,21 +22,24 @@
+    # send data through SPI
+    rxbuf = dev.write(txbuf)
+    value = ((rxbuf[1] & 0x03) << 8) | rxbuf[2]
+    print value
```
Ok, this was one of the two conflicts between my fixes and your first version of the PR. You've picked your version here, but this reverts my fix - and will produces Sonar hits. Should be using `print()` form, like all other places we use this function.

Mani-Sadhasivam commented:
@alext-mkrs Done! Lack of sleep ;-)

arfoll commented:
Sorry was at FOSDEM2018 - this is great, merged.
